core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -84,12 +85,15 @@ private DBWriter() {
             if (media != null) {
                 Log.i(TAG, String.format("Requested to delete FeedMedia [id=%d, title=%s, downloaded=%s",
                         media.getId(), media.getEpisodeTitle(), String.valueOf(media.isDownloaded())));
-                boolean result = false;
                 if (media.isDownloaded()) {
                     // delete downloaded media file
                     File mediaFile = new File(media.getFile_url());
                     if (mediaFile.exists()) {
-                        result = mediaFile.delete();
+                        if (!mediaFile.delete()) {
+                            Toast.makeText(context, "Unable to delete file. Rebooting the device could help.",
+                                    Toast.LENGTH_LONG).show();
```

Comment:
I am not sure from which threads this is called (auto deletion?). Might lead to crashes if it's not the UI thread. Additionally, I never got into a situation where this is actually happening. I think we should finish #2339 first and see if it already fixes the problems. Additionally, DBWriter does not look like a class that should do UI stuff